File: CookingWebsite/CookingWebsite/ClientApp/src/app/pages/recipe-create/update-recipe-dto.ts
import { RecipeIngredientDto } from "../../Dtos/recipe-ingredient-dto";
import { AddRecipeStepDto } from "./add-recipe-step-dto";

export interface UpdateRecipeDto {
  id: number;

  image: File;

  title: string;
  description: string;

  tags: string[];

  cookingTime: number;
  personsCount: number;

  ingredients: RecipeIngredientDto[];
  steps: AddRecipeStepDto[];
}

File: CookingWebsite/CookingWebsite.Application/Recipes/RecipeDtos/UpdateRecipeDto.cs
using CookingWebsite.Application.Files;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CookingWebsite.Application.Recipes.RecipeDtos
{
    public class UpdateRecipeDto
    {
        public int Id { get; set; }
        public File Image { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int CookingTime { get; set; }
        public int PersonsCount { get; set; }
        public List<RecipeIngredientDto> Ingredients { get; set; }
        public List<RecipeStepDto> Steps { get; set; }
        public List<RecipeTagDto> Tags { get; set; }
    }
}
